src/contexts/tfvccontext.ts

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+"use strict";
+
+import { IRepositoryContext, RepositoryType } from "./repositorycontext";
+import { Tfvc } from "../tfvc/tfvc";
+import { Repository } from "../tfvc/repository";
+import { IWorkspace } from "../tfvc/interfaces";
+import { RepoUtils } from "../helpers/repoutils";
+
+export class TfvcContext implements IRepositoryContext {
+    private _tfvcFolder: string;
+    private _gitParentFolder: string;
+    private _tfvcRemoteUrl: string;
+    private _isSsh: boolean = false;
+    private _isTeamServicesUrl: boolean = false;
+    private _isTeamFoundationServer: boolean = false;
+    private _teamProjectName: string;
+
+    constructor(rootPath: string) { //, gitDir?: string) {
+        this._tfvcFolder = rootPath;
+    }
+
+    //Need to call tf.cmd to get TFVC information (and constructors can't be async)
+    public async Initialize(): Promise<Boolean> {
+        //TODO: This will call tf.cmd workfold to determine if we're a TFVC repo
+        //If that succeeds, set _tfvcFolder, _tfvcRemoteUrl and the flags.  That
+        //call will also return the teamProject name for us.
+        try {
+            const tfvc: Tfvc = new Tfvc();
+            const repo: Repository = tfvc.Open(this._tfvcFolder);
```

Comment:
Yeah, we should consider.  I think we should create a few more commands and re-visit.  Once we have a few more commands, I hope it'll be clearer where we should be doing it.  Plus, we need to consider passing the credentials to the Tfvc commands.